pkg/mdformatter/linktransformer/link.go

```diff
@@ -180,6 +190,20 @@ func NewValidator(ctx context.Context, logger log.Logger, linksValidateConfig []
 	v.c.OnError(func(response *colly.Response, err error) {
 		v.rMu.Lock()
 		defer v.rMu.Unlock()
+		// Colly stores requests in cache, even when they have errors.
+		// So need to delete requests with errors from cache database.
+		if v.storage != nil {
+			h := fnv.New64a()
+			_, writeErr := h.Write([]byte(response.Ctx.Get(originalURLKey)))
+			if writeErr != nil {
+				v.remoteLinks[response.Ctx.Get(originalURLKey)] = errors.Wrapf(writeErr, "%q could not build requestID %v", response.Request.URL.String(), response.StatusCode)
+			}
+			delErr := v.storage.DeleteRequest(h.Sum64())
```

Comment:
I wonder if it makes sense to use their interface if we have to do all those things manually.. Wouldn't be better to enhance our storage with API like error and store the status code etc too? Knowing fast that something is 404 is also coachable, no? (with maybe different validity)

Reply:
Yup. Was kind of thinking about the same thing, maybe would be better to just do this on our own and not pass any special storage to Colly. 🙂